File: TypeScipt_Linker/src/configuration/configuration.ts
/**
 * Represents the configuration of the application.
 */
export class Configuration {

    /**
     * The local path to the repository.
     */
    repoPath: string = '';

    /**
     * The JSON-RPC 2.0 methods this application supports.
     */
    supportedRPCMethods: string[] = [
        "MethodLookUp",
        "LanguageLookUp",
        "CodePathLookUp",
        "SetPath"
    ];

    /**
     * The programming languages this application supports.
     */
    supportedLanguages:string[] = [
        "TypeScript"
    ]

    folderPath:string = '';
}

File: CSharp_Linker/CSharp_Linker/Configuration.cs
namespace CSharp_Linker
{
    /// <summary>
    /// This class represents the configuration of the application.
    /// </summary>
    public static class Configuration
    {
        /// <summary>
        /// The path to the .csproj file.
        /// </summary>
        public static string CsProjPath { get; set; } = String.Empty;

        /// <summary>
        /// The path to the Repository where the application data is that has to be linked.
        /// </summary>
        public static string RepoPath { get; set; }

        /// <summary>
        /// The supported programming language of the application.
        /// </summary>
        public static List<string> SupportedLanguages { get; } = new List<string> { "CSharp" };

        /// <summary>
        /// The supported methods of the application.
        /// </summary>
        public static List<string> SupportedMehtods { get; } = new List<string>()
                {
                    "MethodLookUp",
                    "LanguageLookUp",
                    "CodePathLookUp",
                    "SetPath"
                };


    }
}

